src/main/java/com/ericsson/ei/config/HttpSessionConfig.java

```diff
@@ -0,0 +1,16 @@
+package com.ericsson.ei.config;
+
+import java.time.Duration;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.session.data.mongo.JdkMongoSessionConverter;
+import org.springframework.session.data.mongo.config.annotation.web.http.EnableMongoHttpSession;
+
+@EnableMongoHttpSession(maxInactiveIntervalInSeconds = 1200, collectionName="sessions")
+public class HttpSessionConfig {
+
+    @Bean
+    public JdkMongoSessionConverter jdkMongoSessionConverter() {
+        return new JdkMongoSessionConverter(Duration.ofMinutes(30));
```

Comment:
Shouldn't these values (1200  and 30) be set from properties file?

Reply:
The one in JdkMongoSessionConverter could be but not the ones in EnableMongoHttpSession annotation. It does not support getting values from properties. If i type "${my.expiration.interval}" it will not accept a String and if i convert it to int using Integer.parseInt() it will tell me that it must be a constant expression. So it appears the annotation does not support getting values from properties. I don't know how to work around this. The value in JdkMongoSessionConverter is called maxInactiveInterval just like the one in annotation but I'm not sure what it does since the parameter that controls the session expiration is the one in annotation.

Reply:
It can be done in similar way as here:
https://github.com/spring-projects/spring-session/issues/114